### PR TITLE
[14.0][ADD] mrp_bom_hierarchy

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
MRP BoM Hierarchy
====

Make it easy to navigate through the BoM hierarchy:

- This module extends the functionality of Bill of Materials
to support users to better maintain the BoM hierarchy.

- The user can navigate from the tree view to child's BoM or parent's BoM,
or to the product's BoM components with a single click.

### CODE REVIEW

File: mrp_bom_hierarchy/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2015-22 ForgeFlow S.L. (https://www.forgeflow.com)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+{
+    "name": "MRP BoM Hierarchy",
+    "summary": "Make it easy to navigate through BoM hierarchy.",
+    "version": "14.0.1.0.0",
+    "author": "forgeFlow, Odoo Community Association (OCA)",
```
Comment (LoisRForgeFlow) [inline mrp_bom_hierarchy/__manifest__.py]: Capital F

File: mrp_bom_hierarchy/model/mrp_bom.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    )
+    def _compute_internal_reference(self):
+        for bom in self:
+            bom.product_default_code = (
+                bom.product_id.default_code
+                or bom.product_id.product_tmpl_id.default_code
+                or bom.product_tmpl_id.default_code
+            )
+
+    bom_lines = fields.One2many("mrp.bom", compute="_compute_bom_lines")
```
Comment (LoisRForgeFlow) [inline mrp_bom_hierarchy/model/mrp_bom.py]: better name would be `child_bom_ids` IMO, current `bom_lines` name it is really misleading.

File: mrp_bom_hierarchy/model/mrp_bom.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+    bom_ids = fields.One2many("mrp.bom", compute="_compute_bom_ids")
```
Comment (LoisRForgeFlow) [inline mrp_bom_hierarchy/model/mrp_bom.py]: Same here, the name is really misleading, I suggest `parent_bom_ids`

**Review by LoisRForgeFlow — COMMENTED** [review]
Nice refactor in general, a couple of comments below

**Comment by MiquelRForgeFlow** [comment]
Comments attended.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Review by andreampiovesana — APPROVED** [review]
go
is slow with 30.000 boms using filter 
- has parent BoMs
- has child BoMs

**Review by JordiBForgeFlow — APPROVED** [review]

**Pull request merged**